### ISSUE TITLE
Remove QuadratureCache from step-62

### ISSUE THREAD

**Issue by dangars** [dealii/dealii#10479]
In step-62 a frequency sweep is done to calculate the spectrum of an elastic structure.

The mass and stiffness matrices are the same for all the frequency steps. We store them 
 in the `QuadratureCache` buffer in order to improve the assembly speed. This has the following advantages/disadvantages:
- Advantange: Assembly is faster
- Disadvantage: It takes lot of memory. The amount of memory is very large because the data for all the quadrature points has to be stored. This can be very large in 3D for high order elements.

@jppelteret proposed an improvement for the assembly #10435 . This improves the assembly speed by a factor of 7. With this change `QuadratureCache` does not offer any more a real benefit.

Should I remove `QuadratureCache` from step-62? The only reason to keep it, would be for pedagogical reasons.

@bangerth  and @jppelteret, before I proceed, could let me know your opinion?

**Comment by jppelteret** [comment]
Yes, that looks like it must store a lot of data, and could be a bottleneck for scaling up the problem size. What is the approximate difference in speed when using it versus the standard non-cached assembly approach? If its just a few percent, then I think that removing it to further simplify the code would be sufficient justification for the change.

**Comment by jppelteret** [comment]
The other thing to remember is that the original implementation will be memorialised in the 9.2 release documentation. So, if the change does get made, then its always possible to direct someone's attention to that version if necessary. The "possibilities for extension" part could also be amended to suggest this (or some other) caching mechanism in cases where its needed (e.g. when using expensive constitutive laws). So the concept need not be lost entirely, even if the code itself is adjusted.

**Comment by dangars** [comment]
In step-62 the assembly takes 27% of the total time (total_time = 0.467s). In step-62 there is only 12000 dofs.

But the code of step-62 can be used for more complex structures. For 1M dofs, the assembly takes 2.5% of the total time, this is because a direct solver is used. It is not easy to develop an iterative solver for the equation of step-62.

I'll remove `QuadratureCache` but I'll discuss `QuadratureCache` in the extensions section as you suggested. That is a good idea, that way the code won't be lost. I'll discuss the advantages and disadvantages.
